pkg/pool-linear/contracts/usdplus/UsdPlusLinearPool.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/IStaticUsdPlusToken.sol";
+
+import "../LinearPool.sol";
+
+contract UsdPlusLinearPool is LinearPool {
+    IStaticUsdPlusToken private immutable _wrappedToken;
+
+    constructor(
+        IVault vault,
+        string memory name,
+        string memory symbol,
+        IERC20 mainToken,
+        IERC20 wrappedToken,
+        uint256 upperTarget,
+        uint256 swapFeePercentage,
+        uint256 pauseWindowDuration,
+        uint256 bufferPeriodDuration,
+        address owner
+    )
+        LinearPool(
+            vault,
+            name,
+            symbol,
+            mainToken,
+            wrappedToken,
+            upperTarget,
+            swapFeePercentage,
+            pauseWindowDuration,
+            bufferPeriodDuration,
+            owner
+        )
+    {
+        _require(address(mainToken) == IStaticUsdPlusToken(address(wrappedToken)).mainToken(), Errors.TOKENS_MISMATCH);
+        _wrappedToken = IStaticUsdPlusToken(address(wrappedToken));
+    }
+
+    function _getWrappedTokenRate() internal view override returns (uint256) {
+        uint256 rate = _wrappedToken.rate();
```

Comment:
If we wanted to make this pool generically EIP4626 compatible, we could use function signature assetsPerShare() here and in the wrapper instead of rate() and scale the number appropriately

I think that function name is unlikely to change at this point 

https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4626.md

https://ethereum-magicians.org/t/eip-4626-yield-bearing-vault-standard/7900/46

Reply:
Additionally would require changing the wrapper mainToken() function to underlying()

Reply:
Here is a WIP branch that generalizes for 4626: https://github.com/balancer-labs/balancer-v2-monorepo/compare/usdplus-erc4626-pool

Reply:
Looks good, 2 methods on StaticUsdPlus already may be renamed to be compatible.
But here we need fix rate issue corresponding to current assetsPerShare docs: https://github.com/ovnstable/balancer-v2-monorepo/commit/1e32d133075cf7eac88bb448d03c6393ce41e834